File: C++/20_SelectionSort.cpp
#include <iostream>
using namespace std;

void swap(int *a, int i, int j){
    *(a+i) = *(a+i)^*(a+j); 
    *(a+j) = *(a+i)^*(a+j); 
    *(a+i) = *(a+i)^*(a+j); 
}

int main(void){
    int n; 
    cout<<"Enter the size of array: ";
    cin>>n;
    int arr[n];
    for (int i = 0; i < n; i++){
        cin>>arr[i];
    }
    cout<<"Before Sorting the Array: \n";
    for (int i = 0; i<n; i++)
        cout<<arr[i]<<"\t";

    cout<<"\n\n~~~~~~~~~Selection Sort~~~~~~~~~\n\n";

    for (int i = 0; i<n-1; i++){
        int min = i;
        for (int j = i+1; j<n; j++){
            if (arr[min]>arr[j])
                min = j;
        }
        if (min!=i)
            swap(arr, i, min);
    }

    cout<<"After Sorting the Array: \n";
    for (int i = 0; i<n; i++)
        cout<<arr[i]<<"\t";
       
    return 0;
}

File: Python/20_SelectionSort.py
def selectionSort(arr, n):
    for i in range(n-1):
        minidx = i
        for j in range(n):
            if arr[j]>arr[minidx]:
                minidx = j
        if minidx!=i:
            arr[minidx], arr[i] = arr[i], arr[minidx]


if __name__ == '__main__':
    n = int(input("Enter the number of elements in array:  "))
    a = []
    for i in range(n):
        a.append(int(input(f"Enter element {i}: ")))
    print("Array before Sorting: ",a)
    selectionSort(a, n)
    print("Array after Sorting: ",a)